lib/vmdb/loggers.rb

```diff
@@ -31,10 +31,18 @@ def self.apply_config(config)
     end
 
     def self.create_logger(log_file_name, logger_class = VMDBLogger)
-      log_file = ManageIQ.root.join("log", log_file_name)
-      logger_class.new(log_file).tap do |logger|
-        logger.extend(ActiveSupport::Logger.broadcast($container_log)) if $container_log
-        logger.extend(ActiveSupport::Logger.broadcast($journald_log))  if $journald_log
+      log_file_name = Pathname.new(log_file_name) unless log_file_name.kind_of?(Pathname)
+
+      if $container_log
+        create_null_logger(log_file_name, logger_class).tap do |logger|
+          logger.extend(ActiveSupport::Logger.broadcast($container_log))
+        end
+      elsif $journald_log
+        create_null_logger(log_file_name, logger_class).tap do |logger|
+          logger.extend(ActiveSupport::Logger.broadcast($journald_log))
+        end
+      else
+        logger_class.new(ManageIQ.root.join("log", log_file_name))
       end
```

Comment:
```suggestion
      if (log_broadcast_target = $container_log || $journald_log)
        create_null_logger(log_file_name, logger_class).tap do |logger|
          logger.extend(ActiveSupport::Logger.broadcast(log_broadcast_target))
        end
      else
        logger_class.new(ManageIQ.root.join("log", log_file_name))
      end
```

may want to change `create_null_logger` to take the extra parameter and extend it in the method.